src/map/battle.cpp

```diff
@@ -4287,6 +4296,69 @@ static void battle_calc_multi_attack(struct Damage* wd, struct block_list *src,s
 	}
 }
 
+/**
+ * Calculates the percentual attack modificator (ATKpercent) based on status changes
+ * These bonuses are added together and the percent is applied to the damage before the defense reduction in pre-renewal
+ * In renewal this simply sets the base skillratio before the actual skill ratio of the skill used is added
+ * This bonus works as a separate unit to the rest (e.g., if one of these is not applied to a skill, then we know none are)
+ * Do not add additional status changes here unless they are confirmed to use ATKpercent
+ * @param bl: Object to calc atkpercent for
+ * @param skill_id: Skill used by object
+ * @param sc: Object's status change information
+ * @return atkpercent with cap_value(watk,0,USHRT_MAX)
+ */
+static unsigned short battle_get_atkpercent(struct block_list* bl, uint16 skill_id, status_change* sc)
+{
+	int atkpercent = 100;
```

Comment:
I'd move this after the switch below. No need to define if it's never utilized.

Reply:
In C you needed to define all variables at the beginning of a {} section, but I guess it's C++ now, ok sure.

Reply:
Fixed.